google/cloud/aiplatform/training_jobs.py

```diff
@@ -66,6 +66,288 @@
 )
 
 
+class TrainingJob(base.AiPlatformResourceNoun):
```

Comment:
This class seems to implement the entire interface so it's a concrete class which gives the impression that it can and should be used. That is not our intention. I think the fastest way to resolve this is:

1. `run_job` should be private `_run_job`
2. `run` should be [abstractmethod](https://docs.python.org/3/library/abc.html#abc.abstractmethod) so all child classes are required to implement it. The expectation is that `run` calls `_run_job`. That will make this class an ABC.
3. Exception strings specific to the child class should be abstract properties which child classes implement. ie:
```python
@property
@abc.abstractmethod
def _model_upload_fail_string(self) -> str:
    pass
```
4. This class should be a private class since it's only consumed in this module.

Reply:
all done